WPMedia/arc-themes-components
# TBRANDS 65 add carousel indicators

## Ticket

- [TBRANDS-65](https://arcpublishing.atlassian.net/jira/software/c/projects/TMEDIA/boards/875?modal=detail&selectedIssue=TBRANDS-65&assignee=5e387d6a1b1d910e5dfd7a9b)

## Description

Add product gallery with thumbnail and dots layout 

## Acceptance Criteria

This should be an additive update to the Carousel component

New options to the Carousel to have indicators either none,dots,thumbnails with none being the default

## Test Steps

1. Checkout branch - `git checkout TBRANDS-65-carousel-thumbnails`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Look at the dot and thumbnail carousel stories (last) 
5. Ensure that the dots go to the correct slide 
6. Look at the thumbnail one on a smaller screen to see the thumbnail highlight and scroll into view if hidden previously. 

## Dependencies 

- Will implement this in the blocks. If this was merged we could use it in the blocks storybook 
- Feature pack pr https://github.com/WPMedia/arc-themes-feature-pack/pull/351
- Goes with blocks pr https://github.com/WPMedia/arc-themes-blocks/pull/1513

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



---
notes 

# Testing 

- `npm run storybook`
- Go to the carousel story with indicator dots
- Click around the buttons
- See the carousel story for the thumbnail 
- [x] Todo: Show active styles around the active image 
- [x] Todo: Add tests for thumbnail 
- [x] Add cover affecting transparency of the button based on active and non-active image
- [x] Add scroll into view


---
